src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -57,45 +63,106 @@ const SampleChatScreen = () => {
-          onBackPress={goBackFromChatScreen}
-          onThreeDotPress={goToChatInfoScreen}
-          avatar={selectedChannel?.channelPicture}
-          user={selectedChannel?.name}
-          type={ANONYMOUS}
+    <>
+      <StatusBar barStyle="light-content" backgroundColor={colors.anon_primary} />
+
+      <View style={styles.keyboardAvoidingView}>
+        {selectedChannel ? (
+          <ChatDetailHeader
+            onAvatarPress={goToChatInfoScreen}
+            onBackPress={goBackFromChatScreen}
+            onThreeDotPress={goToChatInfoScreen}
+            avatar={selectedChannel?.channelPicture}
+            user={selectedChannel?.name}
+            type={ANONYMOUS}
+            anon_user_info_emoji_code={
+              betterSocialMember &&
+              betterSocialMember[memberChat?.user_id]?.anon_user_info_emoji_code
+            }
+            anon_user_info_color_code={
+              betterSocialMember &&
+              betterSocialMember[memberChat?.user_id]?.anon_user_info_color_code
+            }
+          />
+        ) : null}
+        <FlatList
+          contentContainerStyle={styles.flatlistContainer}
+          style={styles.chatContainer}
+          data={updatedChats}
+          inverted={true}
+          initialNumToRender={20}
+          alwaysBounceVertical={false}
+          bounces={false}
+          onLayout={scrollToEnd}
+          keyExtractor={(item, index) => item?.id || index.toString()}
+          ListFooterComponent={
+            ownerChat ? (
+              <BaseSystemChat
+                componentType="SINGLE"
+                messageSingle={`You’re anonymously messaging ${selectedChannel?.name}.\nThey are still able to block you`}
+              />
+            ) : null
+          }
+          renderItem={renderChatItem}
         />
-      ) : null}
-      <FlatList
-        contentContainerStyle={styles.contentContainerStyle}
-        style={styles.chatContainer}
-        data={updatedChats}
-        inverted={true}
-        initialNumToRender={20}
-        alwaysBounceVertical={false}
-        bounces={false}
-        onLayout={scrollToEnd}
-        keyExtractor={(item, index) => item?.id || index.toString()}
-        renderItem={renderChatItem}
-      />
-      <View style={styles.inputContainer}>
-        <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
+
+        {replyPreview && <ChatReplyPreview type={ANONYMOUS} />}
+        <View style={styles.inputContainer}>
+          <InputMessageV2
+            onSendButtonClicked={sendChat}
+            type={ANONYMOUS}
+            emojiCode={selectedChannel?.rawJson.channel.anon_user_info_emoji_code}
+            emojiColor={selectedChannel?.rawJson.channel.anon_user_info_color_code}
+            username={selectedChannel?.name}
+            onToggleConfirm={moveChatSigned}
+          />
+        </View>
+        <Loading visible={loading} />
       </View>
-    </View>
+    </>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this section seem to be well thought out and are likely to improve the functionality of the component. The addition of the `Loading` component and the `ChatReplyPreview` component, as well as the replacement of `AnonymousInputMessage` with `InputMessageV2`, seem to be beneficial changes. However, it's important to ensure that these new components are thoroughly tested in the context of this parent component.


<!-- This is an auto-generated comment by OSS CodeRabbit -->